### ISSUE TITLE
Overleaf broke firenvim

### ISSUE THREAD

**Issue by RoryNesbitt** [glacambre/firenvim#1416]
### What I tried to do
Write a paper with overleaf, after the editor update

### What happened
Overleaf have developed a new editor, that honestly I have no idea what they changed. When you open firenvim in the new editor it doesn't read the entire contents of the file you're editing and instead gets around 88 lines, starting from wherever your cursor was on the page. When you write the file you overwrite the whole of the document therefore deleting most of it.
Switching to the legacy editor lets your use firenvim as before and works perfectly. I've reported this in the editor feedback, but who knows if they'll see that, or if they'll care. As long as the legacy editor sticks around I'm more interested in knowing how they broke it.

**Comment by tarsobcaldas** [comment]
I'm having a different problem.

When I swicth to another file in Overleaf, it doesn't update correctly. It's just the first file opened when I entered the project

**Comment by euglevi** [comment]
The source legacy editor seems to have disappeared from Overleaf, so now firenvim is officially broken in that webpage. Too bad, given that firenvim was especially useful there.

Anyone willing to have a look at this issue? I am sorry that I do not know enough about HTML or firenvim to contribute.

**Comment by euglevi** [comment]
You are right. Indeed on Chrome firenvim is working fine. Well, I will use Chrome for Overleaf until they support Firefox. Thank you very much for the support.

**Comment by MahdiNazemi** [comment]
For those who need to access the Legacy editor, but no longer see it as an option, try appending `?legacy_source_editor=true` to the URL.

**Comment by emile-cochin** [comment]
Since June 2024, the legacy source editor has been completely disabled, has anyone found a way to use firenvim with overleaf in firefox (without the buffer cutting at 80 something lines)?
